subsys/infinispan/src/main/java/org/commonjava/indy/subsys/infinispan/CacheProducer.java

```diff
@@ -194,66 +173,38 @@ public String getId()
         return "infinispan-caches";
     }
 
-    private String interpolateStrFromStream( InputStream inputStream, String path )
+    private String interpolateFromStream( InputStream inputStream ) throws IOException, InterpolationException
     {
-        String configuration;
-        try
-        {
-            configuration = IOUtils.toString( inputStream );
-        }
-        catch ( IOException e )
-        {
-            throw new RuntimeException( "Cannot read infinispan configuration from : " + path, e );
-        }
+        String configuration = IOUtils.toString( inputStream );
 
         StringSearchInterpolator interpolator = new StringSearchInterpolator();
         interpolator.addValueSource( new PropertiesBasedValueSource( System.getProperties() ) );
 
-        try
-        {
-            configuration = interpolator.interpolate( configuration );
-        }
-        catch ( InterpolationException e )
-        {
-            throw new RuntimeException( "Cannot resolve expressions in infinispan configuration from: " + path, e );
-        }
-        return configuration;
+        return interpolator.interpolate( configuration );
     }
 
-    /**
-     * For the ISPN merging, we should involve at least two different xml config scopes here,
-     * one is from indy self default resource xml, another one is from customer's config xml.
-     *
-     * To prevent the error of EmbeddedCacheManager instances configured with same JMX domain,
-     * ISPN should enable 'allowDuplicateDomains' attribute for per GlobalConfigurationBuilder build,
-     * that will cost more price there for DefaultCacheManager construct and ConfigurationBuilder build.
-     *
-     * Since what we need here is simply parsing xml inputStreams to the defined configurations that ISPN
-     * could accept, then merging the two stream branches into a entire one.
-     * What classes this method uses from ISPN are:
-     * {@link ConfigurationBuilderHolder}
-     * {@link ParserRegistry}
-     * {@link ConfigurationManager}
-     *
-     * @param config
-     * @param path
-     */
-    private void mergedCachesFromConfig( String config, String path )
+    private void defineCaches( String content, String path ) throws IOException
     {
+        InputStream stream = IOUtils.toInputStream( content );
+
         if ( cacheManager == null )
         {
-            cacheManager = new DefaultCacheManager();
+            logger.info( "[ISPN] Define caches, from: {}", path );
```

Comment:
This is important to use the whole stream to construct the cache manager. Otherwise, the global config will be missing (as in previous code).